src/EnergyPlus/CurveManager.cc

```diff
@@ -2797,43 +2797,131 @@ namespace Curve {
     }
 
     Real64 Curve::BtwxtTableInterpolation(EnergyPlusData &state,
-                                          Real64 const Var1,                      // 1st independent variable
-                                          ObjexxFCL::Optional<Real64 const> Var2, // 2nd independent variable
-                                          ObjexxFCL::Optional<Real64 const> Var3, // 3rd independent variable
-                                          ObjexxFCL::Optional<Real64 const> Var4, // 4th independent variable
-                                          ObjexxFCL::Optional<Real64 const> Var5, // 5th independent variable
-                                          ObjexxFCL::Optional<Real64 const> Var6  // 6th independent variable
+                                          const Real64 Var1 // 1st independent variable
     )
     {
         // TODO: Generalize for N-dims
-        Real64 var = Var1;
-        var = max(min(var, this->inputLimits[0].max), this->inputLimits[0].min);
-        std::vector<double> target{var};
-        if (present(Var2)) {
-            var = Var2;
-            var = max(min(var, this->inputLimits[1].max), this->inputLimits[1].min);
-            target.push_back(var);
-        }
-        if (present(Var3)) {
-            var = Var3;
-            var = max(min(var, this->inputLimits[2].max), this->inputLimits[2].min);
-            target.push_back(var);
-        }
-        if (present(Var4)) {
-            var = Var4;
-            var = max(min(var, this->inputLimits[3].max), this->inputLimits[3].min);
-            target.push_back(var);
-        }
-        if (present(Var5)) {
-            var = Var5;
-            var = max(min(var, this->inputLimits[4].max), this->inputLimits[4].min);
-            target.push_back(var);
-        }
-        if (present(Var6)) {
-            var = Var6;
-            var = max(min(var, this->inputLimits[5].max), this->inputLimits[5].min);
-            target.push_back(var);
-        }
+        std::vector<double> target{max(min(Var1, this->inputLimits[0].max), this->inputLimits[0].min)};
```

Comment:
Could have `std::clamp`'d this probably without risking any failures.  We are waiting on doing that in the curve value functions because it was revealing an underlying NaN in the unit tests.  But in other functions, it should be a reasonable thing to use.  I don't think it justifies changing it here unless there is something else to change, and could introduce issues, so probably just :pushpin: